storage/entities/peer.js

```diff
@@ -0,0 +1,307 @@
+
+		this.addField('id', 'number', { filter: ft.NUMBER });
+		this.addField('ip', 'string', { format: 'ip', filter: ft.TEXT });
+		this.addField('wsPort', 'number', {
+			format: 'number',
+			filter: ft.NUMBER,
+		});
+		this.addField('state', 'number', { format: 'number', filter: ft.NUMBER });
+		this.addField('os', 'string', { format: 'string', filter: ft.TEXT });
+		this.addField('version', 'string', { format: 'string', filter: ft.TEXT });
+		this.addField('clock', 'number', { format: 'number', filter: ft.NUMBER });
+		this.addField(
+			'broadhash',
+			'string',
+			{ format: 'string', filter: ft.TEXT },
+			stringToByte
+		);
+		this.addField('height', 'number', {
+			format: 'number',
+			filter: ft.NUMBER,
+		});
+
+		this.SQLs = {
+			selectSimple: this.adapter.loadSQLFile('peers/get_simple.sql'),
+			create: this.adapter.loadSQLFile('peers/create.sql'),
+			update: this.adapter.loadSQLFile('peers/update.sql'),
+			updateOne: this.adapter.loadSQLFile('peers/update_one.sql'),
+			isPersisted: this.adapter.loadSQLFile('peers/is_persisted.sql'),
+		};
+	}
+
+	// eslint-disable-next-line class-methods-use-this
+	get FIELD_SET_SIMPLE() {
+		return FIELD_SET_SIMPLE;
+	}
+
+	/**
+	 * Get one peer
+	 *
+	 * @param {filters.Peer|filters.Peer[]} [filters = {}]
+	 * @param {Object} [options = {}] - Options to filter data
+	 * @param {Number} [options.limit=10] - Number of records to fetch
+	 * @param {Number} [options.offset=0] - Offset to start the records
+	 * @param {fieldSets.Peer} [options.fieldSet='FIELD_SET_SIMPLE'] - Fieldset to choose
+	 * @param {Object} tx - Database transaction object
+	 * @return {Promise.<BasicPeer|ExtendedPeer, Error>}
+	 */
+	getOne(filters, options = {}, tx) {
+		this.validateFilters(filters);
+		this.validateOptions(options);
+		const parsedOptions = _.defaults(
+			{},
+			_.pick(options, ['limit', 'offset', 'fieldSet']),
+			_.pick(this.defaultOptions, ['limit', 'offset', 'fieldSet'])
+		);
+		const mergedFilters = this.mergeFilters(filters);
+		const parsedFilters = this.parseFilters(mergedFilters);
+
+		const params = Object.assign(
+			{},
+			{ limit: parsedOptions.limit, offset: parsedOptions.offset },
+			{
+				parsedFilters,
+			}
+		);
+
+		return this.adapter.executeFile(
+			{
+				[Peer.prototype.FIELD_SET_SIMPLE]: this.SQLs.selectSimple,
+			}[parsedOptions.fieldSet],
+			params,
+			{ expectedResult: 1 },
+			tx
+		);
+	}
+
+	/**
+	 * Get list of peers
+	 *
+	 * @param {filters.Peer|filters.Peer[]} [filters = {}]
+	 * @param {Object} [options = {}] - Options to filter data
+	 * @param {Number} [options.limit=10] - Number of records to fetch
+	 * @param {Number} [options.offset=0] - Offset to start the records
+	 * @param {fieldSets.Peer} [options.fieldSet='FIELD_SET_SIMPLE'] - Fieldset to choose
+	 * @param {Object} tx - Database transaction object
+	 * @return {Promise.<BasicPeer[]|ExtendedPeer[], Error>}
+	 */
+	get(filters = {}, options = {}, tx) {
```

Comment:
`tx` as the last param should also be optional. Please change consecutively `tx` to `tx = null`.